### ISSUE TITLE
Notepad++ prevents Windows restart despite Backup enabled

### ISSUE THREAD

**Issue by piomar123** [notepad-plus-plus/notepad-plus-plus#12723]
### Description of the Issue
When leaving an unsaved document, it's impossible to restart Windows without prior manual closing of Notepad++, even if `Periodic Backup` option is enabled.

### Steps to Reproduce the Issue

1. Ensure Periodic Backup option is enabled (Enable session snapshot and periodic backup)
2. Modify or create a new file but keep it unsaved
3. Try to restart Windows without closing Notepad++
4. Restart is prevented with message "This app is preventing you from restarting"
5. Cancel the restart
6. A widow with the following message is presented:
![image](https://user-images.githubusercontent.com/10479845/209860017-3e531416-8b38-4b14-b93b-491efcb9c6a4.png)
7. Click `No`
8. Close Notepad++
9. Closing is possible without any messages and with keeping unsaved documents thanks to the Periodic Backup functionality.
10. Restarting OS is now not prevented

### Expected Behavior
Restarting OS is possible without manual closing of Notepad++.

### Actual Behavior
App is asking to save all documents.

### Debug Information
```
Notepad++ v8.4.7   (64-bit)
Build time : Nov  1 2022 - 23:31:11
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 21H2
OS Build : 19044.2251
Current ANSI codepage : 1250
Plugins : 
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)
```

### Side note
Probably `WM_QUERYENDSESSION` event or similar related to OS shutdown/logout does not handle Periodic Backup correctly. I'm not familiar yet with the Notepad++ codebase to find the specific cause.

**Comment by ozone10** [comment]
Fixed in Notepad++ v8.4.8, can be closed.

**Comment by piomar123** [comment]
Indeed, v8.4.8 looks fixed. Didn't notice it's already released but still in installer safety delay.
